src/engine/bank03.asm

```diff
@@ -2712,27 +2712,62 @@ Func_d244: ; d244 (3:5244)
 	farcall Func_80ba4
 	jp IncreaseScriptPointerBy2
 
-Func_d24c: ; d24c (3:524c)
-	ld hl, $525e
+ScriptCommand_ShowMultichoiceTextbox_ChooseDeckToDuelAgainst: ; d24c (3:524c)
+	ld hl, .multichoice_menu_args
 	xor a
-	call Func_d28c
+	call ShowMultichoiceTextbox
 	ld a, [wd695]
 	ld c, a
 	set_flag_value EVENT_FLAG_76
 	jp IncreaseScriptPointerBy1
 
-	INCROM $d25e, $d271
+.multichoice_menu_args ;d25e
+	dw $0000 ;NPC title for textbox under menu
+	tx Text03f9 ;text for textbox under menu
+	dw MultichoiceTextbox_ConfigTable_ChooseDeckToDuelAgainst ;location of table configuration in bank 4
+	db $03 ;the value to return when b is pressed
+	dw $d695 ;ram location to return result into
```

Comment:
`$d695` already has a label in wram.asm (`wd695`) but this is also a great opportunity to rename the label to something better. (Same goes for `wd693` and `wd694`.)

Reply:
I have renamed these labels at your suggestion, however wd695 is used in ScriptCommand_StartBattle for non-textbox usage. I would have given this label a union however I am not sure how it is being used in StartBattle.

Reply:
What you could do for now is make the union, and leave one side of the union named `wd695` (for its usage in `ScriptCommand_StartBattle` and only rename the side of the union that is used in this context.

Reply:
Included in most recent commit